dali/core/device_guard_test.cc

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2018, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include "dali/core/dynlink_cuda.h"
+#include "dali/core/cuda_utils.h"
+#include "dali/core/device_guard.h"
+
+namespace dali {
+
+TEST(DeviceGuard, ConstructorWithDevice) {
+  int test_device = 0;
+  int guard_device = 0;
+  int current_device;
+  int count = 1;
+
+  cuInitChecked();
+  cudaGetDeviceCount(&count);
```

Comment:
Wouldn't it be beneficial to error check all the calls as well?
CUDA_CALL(cuda*) and CUDA_CALL(cu*)?

Reply:
Done